### ISSUE TITLE
Dask wrappers/interface for pyFFTW

### ISSUE THREAD

**Issue by jakirkham** [pyFFTW/pyFFTW#151]
Often I use FFTs on large data and am eager to get any speed up I can with it. pyFFTW has been great at delivering really nice performance while also making it very easy to use FFTW thanks to the NumPy style interface.

Though one thing that would be really great is to make it easier still to leverage pyFFTW on data much larger than memory and handle this in parallel. One library that is really fantastic for this is Dask. By allowing one to treat massive files like NumPy arrays and leverage delayed computation, one is able to construct different workflows with large data while keeping focusing on the big picture.

In an attempt to get these two great libraries to play together, @mrocklin, @jcrist, and I worked together on abstracting an FFT interface in Dask ( https://github.com/dask/dask/pull/2093 ) to allow one to easily wrap FFT functions from external libraries that match the NumPy API. While doing this I made sure to test against pyFFTW explicitly. It is trivial for any user to roll these themselves. However, I wondered if it would make sense to included some prewrapped copies of these functions in pyFFTW as part of a new Dask interface for instance.

Would be curious to know your thoughts on this proposal.

**Comment by hgomersall** [comment]
I'm not fully sure of what you're proposing here. Would you suggest an addition to the API, something like `pyfftw.dask`? (also, Dask looks really cool and like its something I should be using!).

**Comment by jakirkham** [comment]
Sorry, was thinking something like `pyfftw.interfaces.dask_numpy_fft` (though we can play with the naming). It would be similar to `pyfftw.interfaces.numpy_fft`, but with Dask support enabled.

**Comment by hgomersall** [comment]
I'm not averse to the idea. Can you enumerate the changes that would be needed?

**Comment by jakirkham** [comment]
Sure. Here are the steps. 

1. Create the `pyfftw.interfaces.dask_numpy_fft` module.
2. Wrap `pyfftw.interfaces.numpy_fft` functions (e.g. `fft = dask.array.fft.fft_wrap(pyfftw.interfaces.numpy_fft.fft)`).
3. Add some tests to exercise these wrappers.

Am more than happy to open a PR with these changes.

**Comment by hgomersall** [comment]
Yeah, do it. Thanks!

**Comment by jakirkham** [comment]
Have added a proposal for adding this Dask interface in PR ( https://github.com/pyFFTW/pyFFTW/pull/154 ).

**Comment by jakirkham** [comment]
Opened a new issue for 2-D/N-D support with Dask in the pyFFTW interface. ( https://github.com/pyFFTW/pyFFTW/issues/175 )
